tests/keras/layers/test_core.py

```diff
@@ -6,161 +6,172 @@
-        if K._BACKEND == "tensorflow":
-            return
-        layer = core.Masking(5)
-        func = K.function([layer.input], [layer.get_output()])
-
-        input_data = np.array([[[1, 1], [2, 1], [3, 1], [5, 5]],
-                              [[1, 5], [5, 0], [0, 0], [0, 0]]],
-                              dtype=np.int32)
-        output = func([input_data])[0]
-        expected = np.array([[[1, 1], [2, 1], [3, 1], [0, 0]],
-                            [[1, 5], [5, 0], [0, 0], [0, 0]]])
-        assert(np.all(output == expected))
+def test_input_output():
+    nb_samples = 10
+    input_dim = 5
+    layer = core.Layer()
 
+    # Once an input is provided, it should be reachable through the
+    # appropriate getters
+    input = np.ones((nb_samples, input_dim))
+    layer.input = K.variable(input)
+    for train in [True, False]:
+        assert_allclose(K.eval(layer.get_input(train)), input)
+        assert_allclose(K.eval(layer.get_output(train)), input)
+
+
+def test_connections():
+    nb_samples = 10
+    input_dim = 5
+    layer1 = core.Layer()
+    layer2 = core.Layer()
+
+    input = np.ones((nb_samples, input_dim))
+    layer1.input = K.variable(input)
+
+    # After connecting, input of layer1 should be passed through
+    layer2.set_previous(layer1)
+    for train in [True, False]:
+        assert_allclose(K.eval(layer2.get_input(train)), input)
+        assert_allclose(K.eval(layer2.get_output(train)), input)
+
+
+def test_base():
+    layer = core.Layer()
+    _runner(layer)
+
+
+def test_masked():
+    layer = core.MaskedLayer()
+    _runner(layer)
+
+
+def test_merge():
+    layer_1 = core.Layer()
+    layer_2 = core.Layer()
+    layer_1.set_input_shape((None,))
+    layer_2.set_input_shape((None,))
+    layer = core.Merge([layer_1, layer_2])
+    _runner(layer)
+
+
+def test_dropout():
+    layer = core.Dropout(0.5)
+    _runner(layer)
+
+
+def test_activation():
+    layer = core.Activation('linear')
+    _runner(layer)
+
+
+def test_reshape():
+    layer = core.Reshape(dims=(10, 10))
+    _runner(layer)
+
+
+def test_flatten():
+    layer = core.Flatten()
+    _runner(layer)
+
+
+def test_repeat_vector():
+    layer = core.RepeatVector(10)
+    _runner(layer)
+
+
+def test_dense():
+    layer = core.Dense(10, input_shape=(10,))
+    _runner(layer)
+
+
+def test_act_reg():
+    layer = core.ActivityRegularization(0.5, 0.5)
+    _runner(layer)
+
+
+def test_time_dist_dense():
+    layer = core.TimeDistributedDense(10, input_shape=(None, 10))
+    _runner(layer)
+
+
+def test_time_dist_merge():
+    layer = core.TimeDistributedMerge()
+    _runner(layer)
+
+
+def test_autoencoder():
+    layer_1 = core.Layer()
+    layer_2 = core.Layer()
+
+    layer = core.AutoEncoder(layer_1, layer_2)
+    _runner(layer)
+
+
+def test_maxout_dense():
+    layer = core.MaxoutDense(10, 10, input_shape=(20,))
+    _runner(layer)
+
+
+@pytest.mark.skipif(K._BACKEND == 'tensorflow', reason="currently not working with TensorFlow")
```

Comment:
It's a nitpick, but there are inconsistent quotes in these files. Please just use `'` everywhere.